### PR TITLE
Add ShortStack package

### PR DESCRIPTION
**Pull request by sebastian-luna-valero**
* [X] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [X] This PR adds a new recipe.
* [X] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by sebastian-luna-valero** [comment]
@bioconda/core please review

File: recipes/shortstack/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://github.com/MikeAxtell/{{ name }}/archive/v{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+
+requirements:
+  run:
+    - perl 5.22.0.1
```
Comment (bgruening) [inline recipes/shortstack/meta.yaml]: please don't pin perl if not really necessary.

File: recipes/shortstack/build.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+sed -i.bak '1 s|^.*$|#!/usr/bin/env perl|g' $SRC_DIR/ShortStack
+cp $SRC_DIR/ShortStack $OUTDIR
+cp $SRC_DIR/README $OUTDIR
+
+# create executables
+BINDIR=$PREFIX/bin
+mkdir -p $BINDIR
+
+ShortStack=$BINDIR/ShortStack
+echo "#! /bin/bash" > $ShortStack;
```
Comment (bgruening) [inline recipes/shortstack/build.sh]: the space should be removed

File: recipes/shortstack/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    - samtools 1.3.1
```
Comment (bgruening) [inline recipes/shortstack/meta.yaml]: do all the other deps really need to be pinned that strict?

File: recipes/shortstack/meta.yaml (hunk 1 above)
Comment (sebastian-luna-valero) [reply recipes/shortstack/meta.yaml]: The program is a `perl` script:
https://github.com/MikeAxtell/ShortStack/blob/master/ShortStack

File: recipes/shortstack/meta.yaml (hunk 3 above)
Comment (sebastian-luna-valero) [reply recipes/shortstack/meta.yaml]: I think so, or that's what I gather from the developer's README:
https://github.com/MikeAxtell/ShortStack/blob/master/README

ping @MikeAxtell

File: recipes/shortstack/meta.yaml (hunk 1 above)
Comment (bgruening) [reply recipes/shortstack/meta.yaml]: Yes, but you don't need to pin the version, isn't it?

Comment (sebastian-luna-valero) [reply recipes/shortstack/meta.yaml]: I had issues when leaving the `perl` version unspecified... can try to reproduce it to check

**Comment by sebastian-luna-valero** [comment]
@bgruening it looks like the pinning can be relaxed.

Please let me know your opinion now.

File: recipes/shortstack/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://github.com/MikeAxtell/{{ name }}/archive/v{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+
+requirements:
+  run:
+    - perl
```
Comment (epruesse) [inline recipes/shortstack/meta.yaml]: pin to 5.*?

File: recipes/shortstack/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    - samtools 1.*
+    - bowtie 1.*
```
Comment (epruesse) [inline recipes/shortstack/meta.yaml]: pin not really necessary, there never will be a 2.0 version of the `bowtie` package given that there is a `bowtie2` package...

File: recipes/shortstack/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    - viennarna 2.*
+
```
Comment (epruesse) [inline recipes/shortstack/meta.yaml]: According to https://github.com/MikeAxtell/ShortStack/blob/master/README you are missing `gzip` and `RNAfold`?

File: recipes/shortstack/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+about:
+  home: https://github.com/MikeAxtell/ShortStack
+  license: GPL-3.0
+  license_family: GPL
+  license_file: LICENSE
+  summary: 'ShortStack: Comprehensive annotation and quantification of small RNA genes'
+
+extra:
+  recipe-maintainers:
+    - sebastian-luna-valero
```
Comment (epruesse) [inline recipes/shortstack/meta.yaml]: Bioconda doesn't really have this setting. It's used in conda-forge to initialize the permissions for the feedstock repositories.

**Comment by epruesse** [comment]
LGTM

**Comment by sebastian-luna-valero** [comment]
@epruesse all valid points, many thanks!

Please let me know your thoughts now.

**Comment by sebastian-luna-valero** [comment]
Will merge now. Thanks both.

**Pull request merged**